[shopsys] use ElasticSearch to get data to readmodel (#1096)
shopsys/shopsys: docs/model/introduction-to-read-model.md

The objects in read model are immutable, read-only by definition, and do not have any behavior.

The read model is implemented in a separate [`shopsys/read-model`](https://github.com/shopsys/read-model) package.

Currently, you can choose between two implementations of `ListedProductViewFacadeInterface` that represents read model:
 - `ListedProductViewElasticFacade` *(default)*

>     - gets data from Elasticsearch

Reviewer: ```suggestion
    - data from Elasticsearch
```

    - faster than getting products from SQL

>     - to use this implementation you need to use `ProductOnCurrentDomainElasticFacade` as well. You can find more about it [here](/docs/model/front-end-product-filtering.md)

Reviewer: ```suggestion
    - to use this implementation you need to use `ProductOnCurrentDomainElasticFacade` as well. You can find more about this topic in [Front-end product filtering](/docs/model/front-end-product-filtering.md)
```

 - `ListedProductViewFacade`

>     - gets data from SQL

Reviewer: ```suggestion
    - data from SQL
```

>     - slower than Elasticsearch, but on the other hand can be used easily on more complex pricing models (for example exact price is calculated with SQL function)

Reviewer: ```suggestion
    - slower than Elasticsearch, but can be easily used for complex pricing models that calculate prices by SQL function
```

>     - to use this implementation you need to use `ProductOnCurrentDomainFacade` as well. You can find more about it [here](/docs/model/front-end-product-filtering.md)

Reviewer: ```suggestion
    - to use this implementation you need to use `ProductOnCurrentDomainFacade` as well. You can find more about this topic in [Front-end product filtering](/docs/model/front-end-product-filtering.md)
```
